frontend/packages/metal3-plugin/src/components/baremetal-hosts/BareMetalHostsPage.tsx

```diff
@@ -52,10 +52,33 @@ type BareMetalHostsPageProps = {
   hasNodeMaintenanceCapability: boolean;
 };
 
+const getCreateProps = ({ namespace }: { namespace: string }) => {
+  const items: any = {
+    dialog: 'New with Dialog',
+    yaml: 'New from YAML',
+  };
+
+  return {
+    items,
+    createLink: (itemName) => {
+      const base = `/k8s/ns/${namespace || 'default'}/${referenceForModel(BareMetalHostModel)}`;
+
+      switch (itemName) {
+        case 'dialog':
+          return `${base}/~new-dialog`;
```

Comment:
Why the path change? `~new` and `~new/form` is used everywhere across the console codebase. Let's stick to that.

Reply:
I think semantically `~new` YAML editor is not a parent resource of form but an alternative view, so IMO `~new` and `~new-dialog` should be a sibling resource.

Reply:
I'd really prefer not diverging from routes established across console. If the semantics of resource creation routes need changing we should do it across the whole console. If we diverge in this single place it will be up to us to keep it aligned to any potential changes done by console-wide refactoring.

Reply:
reverted